code/game/machinery/dehumidifier.dm

```diff
@@ -0,0 +1,156 @@
+//This magic dehumidifier is a portable water-to-nothing machine, although it does need a charged cell to work.
+//See drain.dm!
+/obj/machinery/dehumidifier
+	use_power = POWER_USE_OFF
+	anchored = 0
+	density = 1
+	icon = 'icons/obj/atmos.dmi'
+	icon_state = "dhum-off"
+	name = "emergency dehumidifier"
+	desc = "A portable dehumidifier with emergency high-volume pump to help keep you dry when it's getting damp... just very slowly."
+	var/obj/item/weapon/cell/cell
+	var/on = 0
+	var/drainage = 1
+	var/last_gurgle = 0
+	atom_flags = ATOM_FLAG_CLIMBABLE
+	clicksound = "switch"
+
+/obj/machinery/dehumidifier/Initialize()
+	..()
+	cell = new/obj/item/weapon/cell/high(src)
+	update_icon()
+
+/obj/machinery/dehumidifier/on_update_icon(var/rebuild_overlay = 0)
+	if(!on)
+		icon_state = "dhum-off"
+	else
+		icon_state = "dhum-on"
+
+	if(rebuild_overlay)
+		cut_overlays()
+		if(panel_open)
+			add_overlay("dhum-open")
+
+/obj/machinery/dehumidifier/examine(mob/user)
+	. = ..(user)
+
+	to_chat(user, "The dehumidifier is [on ? "on" : "off"] and the hatch is [panel_open ? "open" : "closed"].")
+	if(panel_open)
+		to_chat(user, "The power cell is [cell ? "installed" : "missing"].")
+	else
+		to_chat(user, "The charge meter reads [cell ? round(cell.percent(),1) : 0]%")
+
+/obj/machinery/dehumidifier/emp_act(severity)
+	if(stat & (BROKEN|NOPOWER))
```

Comment:
Not sure why you're checking state rather than just EMPing the cell anyway.

Reply:
Blame me for copypasting code I don't understand. I assumed it was some sort of sanity check for broken meaning it can't have a cell or something, but given it's immediately followed by checking if a cell exists that's entirely pointless, so yeah, removing.